src/content/1.7/development/components/grid/tutorials/customize-templates.md

```diff
@@ -3,4 +3,169 @@ title: How to customize the Grid templates?
+
+This is the content of this folder (in 1.7.6.x branch):
+
+```
+Blocks
+├── bulk_actions.html.twig
+├── bulk_actions_select_all.html.twig
+├── EmptyState
+│   ├── _default.html.twig
+│   └── supplier.html.twig
+├── grid_actions.html.twig
+├── pagination.html.twig
+├── Table
+│   ├── empty_row.html.twig
+│   ├── filters_row.html.twig
+│   └── headers_row.html.twig
+└── table.html.twig
+```
+
+The templates have meaningful names so you will spot the reponsibility
+of most of them:
+
+* The **bulk actions** templates are responsible of the rendering of the parent container of the bulk action templates;
+* The **grid actions** templates are responsible of the rendering of the parent container of the grid action templates;
+* All **table** related templates are responsible of the rendering of the Table structure of the Grid (we could name it "Grid" too as you're not forced to use an HTML table to render a Grid!);
+* The **empty state** templates define the rendering of the Grid in case you have no data to render;
+* Finally, the **pagination** template is responsible of... guess what? the rendering of the pagination system! Obvious 😎.
+
+> Customize these template when you want to alter the rendering of the structure of your Grids.
+
+## The Columns Contents and Headers
+
+Most of the time, you want to customize the rendering of the columns. While we can re-order add or delete columns using hooks, the Twig layer allows
+you to alter the rendering of a column at Shop, Grid or even Column level.
+
+It's not a surprise that you will retrieve a template named after the available Column Types provided by the Core of PrestaShop! The Twig layer relies on the information sent by the Grid presenter to select the right Twig template to be rendered and to push the right information.
+
+This is the content of this folder (in 1.7.6.x branch):
+
+```
+Columns
+├── Content
+│   ├── action.html.twig
+│   ├── badge.html.twig
+│   ├── bulk_action.html.twig
+│   ├── category_position.html.twig
+│   ├── data.html.twig
+│   ├── date_time.html.twig
+│   ├── employee_name_with_avatar.html.twig
+│   ├── image.html.twig
+│   ├── link.html.twig
+│   ├── position_handle.html.twig
+│   ├── position.html.twig
+│   ├── severity_level.html.twig
+│   └── toggle.html.twig
+└── Header
+    └── Content
+        ├── action.html.twig
+        ├── default.html.twig
+        └── position_handle.html.twig
+```
+
+The templates of this folder are splitted into two subfolders:
+
+* **Content**: contains the template for each type of Column, by default `data.html.twig` is used;
+* **Header/Content**: contains the template for each type of Column Header (in the current context, the column table header). By default, `default.html.twig` is used;
+
+For each template, you can customize the rendering at different levels.
+There is a simple rule to guess which template will be rendered: `{grid_id}_{column_id}_type_id` where `grid_id` and `column_id` are not required.
```

Comment:
there is more to it.

first it checks for `{grid_id}_{column_id}_{column_type}.html.twig` - use it to customize specific column in specific grid
then `{grid_id}_{column_type}.html.twig` - use it to customize all columns of same type (e.g. `DataColumn`) in specific grid
and last `{column_type}.html.twig` - use to customize all columns of specific type for all grids

Reply:
this is exactly what I have explained: note that the column type is always required, this is why I didn't use curly braces around it.